demo/src/components/colors.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from "react";
+import { getColor, setColor } from "../utils/colors";
+
+const Colors = () => {
+  const [colors, setColors] = useState({
+    snow1: getColor("snow1"),
```

Comment:
**question:** Is there a way to listen for changes to theme preference, so when switching from for instance light to dark the `Colors` component fetches the current colors again?

Reply:
Sure, we should be able to use the `window.watchMedia("prefers-color-scheme: dark")` API and set the CSS custom properties when the preference changes.

So you're saying that you want the color changes to be tied to the browser light / dark mod settings instead of color inputs?

Reply:
This chunk of code handles listening to system color scheme preferences, but I'm not sure what you want to happen in these cases.

```javascript
useEffect(() => {
  const mql = window.matchMedia("(prefers-color-scheme: dark)");

  const onChange = (e) => {
    console.log(`${e.matches ? "dark" : "light"} mode`);
  };

  mql.addEventListener("change", onChange);

  return () => mql.removeEventListener("change", onChange);
}, []);
```

Reply:
Inside the onChange grab the css color values again. 

The case I tested:
- Open site
- Do not pick a color
- Change the prefered theme (inside the dev tools)
- The colors in the picker now does not match color of snow falling

Reply:
I implemented what you described above in the latest commit!

However, now I think we're in a funny spot. If the user updates the snow colors with the inputs, swapping between light and dark mode no longer modifies the colors. I think it's because calling `document.documentElement.style.setProperty()` actually _inlines_ the custom property, which overrides what's provided in `global.css`.

```
<html style="--snow1:#6c180e; --snow2:#917e7e;">
```

I think we should only have one input method for colors to avoid this situation. Whether that be dark or light mode toggle, or the two inputs.

What do you think?

Reply:
I think if the visitor first overrides the colors then they are overwritten.

Reply:
Okay, cool! Then the latest commit should match the functionality that you describe!